cfdm/test/test_read_write.py

```diff
@@ -905,6 +905,36 @@ def test_read_original_filenames(self):
             set((cfdm.abspath(parent_file), cfdm.abspath(external_file))),
         )
 
+    def test_write_omit_data(self):
+        """Test the `omit_data` parameter to `write`."""
+        f = cfdm.example_field(1)
+        cfdm.write(f, tmpfile)
+
+        cfdm.write(f, tmpfile, omit_data="all")
+        g = cfdm.read(tmpfile)
+        self.assertEqual(len(g), 1)
+        g = g[0]
+
+        # Check that the data are missing
```

Comment:
Can we also do a quick `dump` on the re-read field `g` to further check that the missing data doesn't confuse a representation call? I say this mostly because I recall we have previously had a bug with representation when a field had missing data for constructs (see NCAS-CMS/cf-python#326) so having data not present on disk could cause similar issues as a representation edge case that would be good to test against. (I did check manually that `dump` works for all of the example fields when re-read in after being written out with `omit_data="all"`, so the tests should pass, but would be good to include for coverage!)

Reply:
Added in afebbd6 and the previous commit, so marking as resolved.